lib/improver/tests/metadata/test_check_datatypes.py

```diff
@@ -114,29 +113,12 @@ def test_float64_cube_coord_bounds_with_fix(self):
             np.array([(point - 10., point + 10.) for point in x_coord.points])
         )
         expected_cube = self.cube.copy()
-        expected_cube.coord("projection_x_coordinate").points = (
-            expected_cube.coord("projection_x_coordinate").points.astype(
-                np.float64))
         expected_coord = expected_cube.coord("projection_x_coordinate")
         enforce.check_cube_not_float64(self.cube, fix=True)
         self.assertEqual(self.cube, expected_cube)
         self.assertEqual(
             self.cube.coord("projection_x_coordinate"), expected_coord)
```

Comment:
The comparison statement here does not compare datatypes, and the datatype of `expected_coord` points and bounds is "wrong" (in the sense that it is not what we want - luckily the datatype of the coordinate points and bounds on the actual returned cube is correct).  There is quite a lot wrong with the tests in this class, but I've not touched anything more than this and would like to come back to these in a separate PR.